engine/access/rest/events.go

```diff
@@ -0,0 +1,111 @@
+package rest
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/access"
+)
+
+const blockQueryParam = "block_ids"
+const eventTypeQuery = "type"
+
+// getEvents for the provided block range or list of block IDs filtered by type.
+func getEvents(r *request, backend access.API, link LinkGenerator) (interface{}, error) {
+	startHeight := r.getQueryParam(startHeightQueryParam)
+	endHeight := r.getQueryParam(endHeightQueryParam)
+	eventType := r.getQueryParam(eventTypeQuery)
+	blockIDs, err := r.getQueryParams(blockQueryParam)
+	if err != nil {
+		return nil, NewBadRequestError(fmt.Errorf("invalid list of block IDs: %w", err))
+	}
+
+	// if both height and one or both of start and end height are provided
+	if len(blockIDs) > 0 && (startHeight != "" || endHeight != "") {
+		err := fmt.Errorf("can only provide either block IDs or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	// if neither height nor start and end height are provided
+	if len(blockIDs) == 0 && (startHeight == "" || endHeight == "") {
+		err := fmt.Errorf("must provide either block IDs or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	if eventType == "" { // todo(sideninja) validate format with regex in the validation layer
+		err := fmt.Errorf("event type must be provided")
+		return nil, NewBadRequestError(err)
+	}
+
+	if len(blockIDs) > 0 {
```

Comment:
it seems weird to accept either a start/end height range or list of IDs instead of a list of heights. Does it make sense to accept a list of heights as well?